File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    words_set = set()
    prev_word = words[0][0]
    
    for i,w in enumerate(words):
        if w in words_set or w[0] != prev_word:
            return [(i%n)+1, (i//n)+1]
        words_set.add(w)
        prev_word = w[-1]

    return [0,0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let answer = 0;
    
    words.reduce((a, c, i) => {
        answer = answer || ((words.slice(0, i).indexOf(c) !== -1 || a !== c[0]) ? i : answer);
        return c[c.length-1];
    }, "")

    return answer ? [answer%n+1, Math.floor(answer/n)+1] : [0,0];
}